src/transformers/trainer.py

```diff
@@ -1530,6 +1530,10 @@ def train(
             and not self.is_fsdp_enabled
         ):
             self._load_from_checkpoint(resume_from_checkpoint)
+            # In case of repeating the find_executable_batch_size, set `self._train_batch_size` properly
+            state = TrainerState.load_from_json(os.path.join(resume_from_checkpoint, TRAINER_STATE_NAME))
```

Comment:
Don't necessarily enjoy the fact we load it up here just for one value, but it makes sense to keep this metadata in here rather than in the model metadata as that makes no sense, nor does it make sense to have a whole new dataclass/state/thing for us to store it in either

Reply:
Indeed, it's not ideal. How large is it / how long does it take to load? 

Could we protect it behind the `if state.train_batch_size is not None` branch until there's a need for the state later.

Reply:
Not large at all, it's just got a few dataclass entries in it. However we can certainly protect it to reduce IO time

Reply:
Also that's not really possible, because a user should be able to do `auto_find_batch_size` first, and then not require it again if they're resuming from a checkpoint and that checkpoint stored information about the prior run, for us to load in automatically, so thus it's always needed. This is only happening if we resume from checkpoint which should limit it enough,.